test/suites/drift/suite_test.go

```diff
@@ -337,6 +337,111 @@ var _ = Describe("Drift", Label("AWS"), func() {
 		Entry("DetailedMonitoring Drift", "DetailedMonitoring", v1alpha1.AWSNodeTemplateSpec{DetailedMonitoring: awssdk.Bool(true)}),
 		Entry("AMIFamily Drift", "AMIFamily", v1alpha1.AWSNodeTemplateSpec{AWS: v1alpha1.AWS{AMIFamily: awssdk.String(v1alpha1.AMIFamilyBottlerocket)}}),
 	)
+	It("should not continue to drift if a machine fails to register", func() {
```

Comment:
nit: consider renaming to include something about the delete
```suggestion
	It("should delete a replacement machine from drift if it fails to register", func() {
```